### PR TITLE
enterprise webservice tests marked optional

### PR DESCRIPTION
**Pull request by alwin-joseph**
This is to mark all Jakarta Enterprise Web Services(JSR 109) tests as optional as per https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan.

There is also a suggestion to remove webservices12 tests as they are covering pre-EE 8 versions.

There are other tests using  Jakarta XML Web Services (Optional) + Jakarta Web Services Metadata (Optional) in platform tck run that will be marked as optional in separate changes.

### CODE REVIEW

**Comment by scottmarlow** [comment]
Note the [Full Platform specification](https://jakarta.ee/specifications/platform/9/platform-spec-9-SNAPSHOT.html#xml-web-services-optional):
```
2.7.15. XML Web Services (Optional)

Jakarta EE optionally provides full support for both clients of web services as well as web service endpoints. Several Jakarta technologies work together to provide support for web services. Jakarta XML Web Services provides support for web service calls using the SOAP/HTTP protocol. XML Web Services is the primary API for web services and is a follow-on to Jakarta XML-based RPC[1]. Jakarta XML Web Services offers extensive web services functionality, with support for multiple bindings/protocols. Support for Jakarta XML-based RPC has been removed from the Platform as of Jakarta EE 9. See Removed Java Technologies.

Jakarta XML Web Services and Jakarta XML Binding define the mapping between Java classes and XML as used in SOAP calls, and provide support for 100% of XML Schema. The Jakarta SOAP with Attachments provides support for manipulating low level SOAP messages. The Web Services for Jakarta EE specification fully defines the deployment of web service clients and web service endpoints in Jakarta EE, as well as the implementation of web service endpoints using enterprise beans. The XML Web Services Metadata specification defines Java language annotations that make it easier to develop web services. The Jakarta XML Registries support has been removed from the Platform as of Jakarta EE 9. See Removed Java Technologies.
```

While the [Web Profile specification](https://jakarta.ee/specifications/webprofile/9/webprofile-spec-9-SNAPSHOT.html) does not (same as Jakarta XML binding which also is not defined in Web Profile).

I'm thinking that we should remove `javaee_web/javaee_web_profile_optional` for Enterprise webservice tests tests.  Probably the same for Jakarta XML binding tests.

**Comment by alwin-joseph** [comment]
> Note the [Full Platform specification](https://jakarta.ee/specifications/platform/9/platform-spec-9-SNAPSHOT.html#xml-web-services-optional):
> 
> ```
> 2.7.15. XML Web Services (Optional)
> 
> Jakarta EE optionally provides full support for both clients of web services as well as web service endpoints. Several Jakarta technologies work together to provide support for web services. Jakarta XML Web Services provides support for web service calls using the SOAP/HTTP protocol. XML Web Services is the primary API for web services and is a follow-on to Jakarta XML-based RPC[1]. Jakarta XML Web Services offers extensive web services functionality, with support for multiple bindings/protocols. Support for Jakarta XML-based RPC has been removed from the Platform as of Jakarta EE 9. See Removed Java Technologies.
> 
> Jakarta XML Web Services and Jakarta XML Binding define the mapping between Java classes and XML as used in SOAP calls, and provide support for 100% of XML Schema. The Jakarta SOAP with Attachments provides support for manipulating low level SOAP messages. The Web Services for Jakarta EE specification fully defines the deployment of web service clients and web service endpoints in Jakarta EE, as well as the implementation of web service endpoints using enterprise beans. The XML Web Services Metadata specification defines Java language annotations that make it easier to develop web services. The Jakarta XML Registries support has been removed from the Platform as of Jakarta EE 9. See Removed Java Technologies.
> ```
> 

The above references Jakarta XML Web Services (Optional) covered by jaxws tck.

> While the [Web Profile specification](https://jakarta.ee/specifications/webprofile/9/webprofile-spec-9-SNAPSHOT.html) does not (same as Jakarta XML binding which also is not defined in Web Profile).

Looks like it was the same for https://jakarta.ee/specifications/webprofile/8/webprofile-spec-8.html as well.
The "Table 2. Jakarta EE Technologies" in https://jakarta.ee/specifications/platform/9/platform-spec-9-SNAPSHOT.html#a2161 mentions "Y" for Web 

> 
> I'm thinking that we should remove `javaee_web/javaee_web_profile_optional` for Enterprise webservice tests tests. Probably the same for Jakarta XML binding tests.

**Review by gurunrao — COMMENTED** [review]
IMO - We should provide choice for vendor to test optional tech(if vendor wishes to support until its removed from jakarta EE). If Vendor doesn't want to support optional tech, he can pass !javaee_web_profile_optional. Even though Enterprise WS is not required components, if a Vendor is already supporting Enterprise WS is the scenario, I have in mind.
Enterprise WS might not be part of Web profile, but as Web profile spec says 

> "Web Profile products may support some of the technologies present in the full Jakarta EE Platform and not already listed in Required Components, consistently with their compatibility requirements."

**Comment by scottmarlow** [comment]
I'm not sure of a reason to remove the webservices12 tests now.

File: src/com/sun/ts/lib/harness/keyword.properties [hunk 1]
```diff
@@ -240,28 +240,28 @@ com/sun/ts/tests/servlet/ee/platform/deploy/ejbref/single = javaee_optional java
 #com/sun/ts/tests/jaxws = jaxws webservices javaee jaxws_web_profile javaee_web_profile_optional
 #com/sun/ts/tests/jaxws/se = jaxws webservices javaee jaxws_web_profile javaee_web_profile_optional seonly
 
-com/sun/ts/tests/webservices12 = webservices javaee
+com/sun/ts/tests/webservices12 = webservices javaee_optional
```
Comment (scottmarlow) [inline src/com/sun/ts/lib/harness/keyword.properties:243]: +1 looks right to me.

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**